src/chart/tooltip.js

```diff
@@ -0,0 +1,87 @@
+import d3 from 'd3';
+import dataJoinUtil from '../util/dataJoin';
+import {noop} from '../util/fn';
+
+export default function() {
+
+    var labelWidth = 50,
+        decorate = noop;
+
+    var items = [
+        ['datum:', function(d) { return d.datum; }]
```

Comment:
Unsure about the structure, I don't like unnamed arguments so I would prefer to see an `{label:  () =>, value: () => }[]` but I also like the brevity. 

Maybe it could be done similar to how d3 does it on ordinal scales e.g. 

```js
  tooltip()
    .labels([a, b, c])
    .values([a, b, c])
```